### PR TITLE
[weather.ha] 0.0.5b

### PR DESCRIPTION
**Pull request by Eugeniusz-Gienek**
### Description

The plugin which is aimed to get the weather from the user's Home Assistant server.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: weather.ha/lib/ha_weather.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                if r.status_code == 401:
+                    show_dialog(__addon__.getLocalizedString(30011)) #Error 401: Check your token
+                elif r.status_code == 405:
+                    show_dialog(__addon__.getLocalizedString(30012)) #Error 405: Method not allowed
+                elif r.status_code == 200:
+                    return r
+            except:
+                log('Status code error is: ' + str(r.raise_for_status() if r else 'unknown' ))
+            retry += 1
+            sleep(RETRY_DELAY_S)
```
Comment (basrieter) [inline weather.ha/lib/ha_weather.py:114]: Please don't use `sleep` as it prevents Kodi from shutting down the script. Use the `MyMonitor.waitForAbort(10)` like the other weather add-ons do (https://github.com/xbmc/repo-scripts/blob/26fc0ca904c51d0078b8193c571356dd30b417a8/weather.multi/lib/weather.py#L104C37-L104C54)

File: weather.ha/lib/ha_weather.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+import time
+from datetime import datetime
+import iso8601
+
+from xbmc import log as xbmc_log
+
+from .util import *
+
+from time import sleep
```
Comment (basrieter) [inline weather.ha/lib/ha_weather.py:23]: Don't use sleep in the add-ons.

File: weather.ha/lib/ha_weather.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                if r.status_code == 401:
+                    show_dialog(__addon__.getLocalizedString(30011)) #Error 401: Check your token
+                elif r.status_code == 405:
+                    show_dialog(__addon__.getLocalizedString(30012)) #Error 405: Method not allowed
+                else:
+                    return r
+            except:
+                log('Status code error is: ' + str(r.raise_for_status() if r else 'unknown' ))
+            retry += 1
+            sleep(RETRY_DELAY_S)
```
Comment (basrieter) [inline weather.ha/lib/ha_weather.py:135]: Same here. Don't use `sleep()`

**Review by basrieter — CHANGES_REQUESTED** [review]
Please see my comments.

File: weather.ha/forecasts.example.yaml
[the remark this answers is not in the document]
Comment (Eugeniusz-Gienek) [reply weather.ha/forecasts.example.yaml:1]: Yes, in Home Assistant - in order for the plugin to work you have to put it in HA.

**Review by Eugeniusz-Gienek — COMMENTED** [review]
Fixed the "sleep" issue

**Review by Eugeniusz-Gienek — COMMENTED** [review]
Fix timing to match other weather plugins

**Review by basrieter — APPROVED** [review]

**Pull request merged**